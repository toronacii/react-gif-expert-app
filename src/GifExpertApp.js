import React, { useState } from 'react'

const GifExpertApp = () => {

  const [categories, setCategories] = useState(['One Punch Man', 'Samuray X', 'Dragon Ball'])
  const handleAdd = () => {
    const newCategory = getPromptNewCategory();
    if (newCategory) {
      setCategories([...categories, newCategory]);
    }
  }

  return (
    <div>
      <h2>GifExpertApp</h2>
      <hr />
      <button onClick={handleAdd}>Add</button>
      <ol>{
        categories.map(category => {
          return <li key={category}>{category}</li>
        })
      }</ol>
    </div>
  )

  function getPromptNewCategory() {
    let category;
    do {
      category = prompt('Añade una categoría')?.trim();
      if (category && categories.includes(category)) {
        alert('La categoría ya existe!');
        category = null;
      }
    } while (!category && category !== undefined);
    return category;
  }
}

export default GifExpertApp;
