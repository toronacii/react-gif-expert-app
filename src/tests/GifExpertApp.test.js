import { shallow } from 'enzyme';
import React from 'react';
import GifExpertApp from '../GifExpertApp';

describe('Tests on `GifExpertApp` component', () => {
  let wrapper = _createWrapper();

  beforeEach(() => {
    wrapper = _createWrapper();
  });

  test('GifExpertApp should be created', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should shows categories', () => {
    const categories = ['Category 1', 'Category 2'];
    wrapper = _createWrapper(categories);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });

  function _createWrapper(categories = []) {
    return shallow(<GifExpertApp defaultCategories={categories} />);
  }
});