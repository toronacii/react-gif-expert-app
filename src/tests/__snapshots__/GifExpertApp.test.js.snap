// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on \`GifExpertApp\` component GifExpertApp should be created 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`Tests on \`GifExpertApp\` component should shows categories 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Category 1"
      key="Category 1"
    />
    <GifGrid
      category="Category 2"
      key="Category 2"
    />
  </ol>
</div>
`;
